run/awake_start

```diff
@@ -27,7 +27,42 @@ function ping {
       curl -s -H "Authorization: Bearer $TESLAFI_API_TOKEN" "https://www.teslafi.com/feed.php?command=wake_up"
       # No need to contact TeslaFi too often to prevent API lockout (max 3 calls/min).
       sleep 600
-    fi    
+    fi
+
+    if [[ -n "${TESSIE_API_TOKEN:+x}" ]]
+    # Use Tessie API to keep car awake
+    then
+        http_response=$(curl -s -w "%{http_code}" \
+            -H "Authorization: Bearer $TESSIE_API_TOKEN" \
+            -H "Accept: application/json" \
+            -H "User-Agent: github.com/marcone/teslausb" \
+            "https://api.tessie.com/$TESSIE_VIN/wake")
+        
+        # Extract HTTP status code
+        http_status="${http_response: -3}"
+        
+        # Extract response body (excluding status code)
+        response_body="${http_response:0:${#http_response}-3}"
+        
+        # Check if HTTP status code is not 200 (OK)
+        if [ "$http_status" != "200" ]; then
+            # Log an error
+            log "Failed to wake car with Tessie API. Check your TESSIE_API_TOKEN and TESSIE_VIN are set correctly. If they are, your car may not have a good connection. Retrying in 5 minutes..."
+        else
+            # Check if the response is in JSON format
+            if ! jq -e . >/dev/null 2>&1 <<<"$response_body"; then
+                log "Failed to wake car with Tessie API. Retrying in 5 minutes. Response data: $response_body"
+            else
+                # Parse JSON and check if "result" is true
+                result=$(jq -r '.result' <<<"$response_body")
+                if [[ "$result" != "true" ]]; then
+                    log "Failed to wake car with Tessie API. Retrying in 5 minutes. Response data: $response_body"
+                fi
+            fi
+        fi
+        
+        sleep 300
```

Comment:
Should this wait less than 5 minutes if a failure occurs? Wouldn't the car most likely have gone to sleep after waiting 5 minutes if this was being relied upon to keep it awake and it just failed?

Reply:
The thought here was not to spam the Tessie API, but I see your point so I'll reduce to 1min in any non-success case. Since if its not keeping it awake then its going to be asleep anyway so won't be hitting the API much.